config/configmapprovider/file_watch.go

```diff
@@ -0,0 +1,81 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmapprovider
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+)
+
+var errNoOp = errors.New("no change")
+
+// watchFile sets up a watch on a filename to detect changes and calls onChange() with a suitable ChangeEvent.
+// The watch is cancelled when the context is done.
+func watchFile(ctx context.Context, filename string, onChange func(*ChangeEvent)) error {
```

Comment:
This does not need to be in a separate file. Consider to have it in "file.go"

Reply:
done